lib/Util/Notices/UnauthorizedEmailNotice.php

```diff
@@ -0,0 +1,112 @@
+      $button = $this->wp->x('Update my Default sender email address', 'Please reuse the current translation of “Default sender”');
+      $button = Helpers::replaceLinkTags("[link]{$button}[/link]", 'admin.php?page=mailpoet-settings', ['class' => 'button button-secondary']);
+      $buttons .= "<p>$button</p>";
+    }
+    if (!empty($validation_error['invalid_confirmation_address'])) {
+      $button = $this->wp->x('Update my Sign-up Confirmation email address', 'Please reuse the current translation of “Sign-up Confirmation”');
+      $button = Helpers::replaceLinkTags("[link]{$button}[/link]", 'admin.php?page=mailpoet-settings#signup', ['class' => 'button button-secondary']);
+      $buttons .= "<p>$button</p>";
+    }
+    return $buttons;
+  }
+
+  private function getAuthorizationLink($validation_error) {
+    if (!empty($validation_error['invalid_sender_address']) && !empty($validation_error['invalid_confirmation_address'])) {
+      $authorize_link = $this->wp->x('Authorize %email and %email2', 'Link for user to authorize their email address');
+      $authorize_link = str_replace('%email', EscapeHelper::escapeHtmlText($validation_error['invalid_sender_address']), $authorize_link);
+      $authorize_link = str_replace('%email2', EscapeHelper::escapeHtmlText($validation_error['invalid_confirmation_address']), $authorize_link);
+    } else {
+      $email = isset($validation_error['invalid_sender_address']) ? $validation_error['invalid_sender_address'] : $validation_error['invalid_confirmation_address'];
+      $authorize_link = $this->wp->x('Authorize %email', 'Link for user to authorize their email address');
+      $authorize_link = str_replace('%email', EscapeHelper::escapeHtmlText($email), $authorize_link);
+    }
+    $authorize_link = Helpers::replaceLinkTags("[link]{$authorize_link}[/link]", 'https://account.mailpoet.com/authorization', ['target' => '_blank']);
+    $html = '<p><b>'. $this->wp->x('OR', 'User has to choose between two options') . '</b></p>';
+    $html .= "<p>$authorize_link</p>";
+    return $html;
+  }
+
+  private function getResumeSendingButton() {
+    $button = '<button class="button button-primary mailpoet-js-button-resume-sending">' . $this->wp->__('Resume sending', 'mailpoet') . '</button>';
+    $script = "
+    <script>
+      jQuery('.mailpoet-js-button-resume-sending').on('click', function() {
+        MailPoet.Ajax.post({
+        api_version: window.mailpoet_api_version,
+        endpoint: 'mailer', action: 'resumeSending'
+        }).done(function() {
+          jQuery('.mailpoet-js-error-unauthorized-emails-notice').slideUp();
+          MailPoet.Notice.success('" . WPFunctions::get()->__('Sending has been resumed.') . "');}
+        ).fail(function(response) {
+          if (response.errors.length > 0) {
+            MailPoet.Notice.error(response.errors.map(function(error) { return error.message }), { scroll: true });
+          }
+        });
+      });
```

Comment:
Uff, I'm not familiar with reasons for this and how other messages are rendered but does this really have to be in PHP? Is there anything preventing us from doing this the "standard" way?

Reply:
I don't like it either :(  I used a system for displaying permanent notices but it can only render HTML. I checked other places and found that we do similar thing here: 
https://github.com/mailpoet/mailpoet/blob/296ad974d8e7433b345df2766309609ab7239d5d/lib/Mailer/Methods/ErrorMappers/MailPoetMapper.php#L92 so I decided to do it same way.

I was thinking about 2 alternative solutions:
- rendering only HTML and adding a js code for resume sending [here](https://github.com/mailpoet/mailpoet/blob/master/assets/js/src/dismissible-notice.jsx) or to some new js file. (preferred current solution since code is together)
- creating an extra react component only for this notice (felt like non system solution and would have to take care e.g. that it disappears when settings are saved etc.)

Do you have some suggestion?

Reply:
I see. It's not in the scope of this feature to refactor the notice system to something nicer but inlining JS like this is really bad. It not only avoids linters but it can also be forgotten when someone will refactor `MailPoet.anything...` API in JS. Don't you think it would be better to extract JS at least? I think it's better to have in two places than to call our JS classes like `MailPoet` from PHP.

Reply:
Good point. I'll extract it.